src/proteingym/base/dataset.py

```diff
@@ -257,6 +276,32 @@ def __repr__(self) -> str:
         lines.append(f"\t\tstructures: {len(self.structures)},")
         lines.append(f"\t\tmsas: {len(self.msas)},")
         lines.append(f"\t\tassay_variables: {len(self.assay_variables)},")
+        if self.publication:
```

Comment:
could this recycle `Publication`'s own `__repr__`? The raising errors feels overdone for a repr method to me